amplify/backend/function/putUser/src/index.py

```diff
@@ -17,21 +26,29 @@ def lambda_handler(event, context):
     query = f"g.V().has('username', '{username}').id();"
     vertexID = db.submit(query).all().result()
 
-    for attribute in userData:
-        newValue = userData[attribute]
-        # Add new value
-        query = f"g.V({vertexID}).property(single, '{attribute}', '{newValue}');"
-        try:
-            if len(vertexID) == 0:
-                result = f"Server failed to find user : {username}"
-                statusCode = 404
+    result = ""
+    statusCode = 500
+    if len(vertexID) == 0:
+        result = f"Server failed to find user : {username}"
+        statusCode = 404
+    elif len(userData) > 0:
+        for attribute in userData:
+            newValue = userData[attribute]
+            query = f"g.V({vertexID}).property(single, '{attribute}', '{newValue}');"
+            # check if attribute is valid
+            if attribute in valid_attributes:
+                try:
```

Comment:
Nit: We don't need to update the `query` and `newValue` variables if the attribute is not valid
```suggestion
            if attribute in valid_attributes:
                newValue = userData[attribute]
                query = f"g.V({vertexID}).property(single, '{attribute}', '{newValue}');"
                try:
```